src/infrastructure/acknowlegement.rs

```diff
@@ -3,7 +3,7 @@ use crate::infrastructure::{ExternalAcks, LocalAckRecord};
 /// Type responsible for handling the acknowledgement of packets.
 pub struct AcknowledgementHandler {
     waiting_packets: LocalAckRecord,
-    their_acks: ExternalAcks,
+    our_acks: ExternalAcks,
```

Comment:
Hrm. It looks like the ExternalAcks struct is conflating storing the remote sequence number with our own calculated version of the ack field. This implementation seems a bit confused.

Reply:
ExternalAcks keeps track of *our* receipt of the last 33 *foreign* packets, and that alone. if anything conflate its LocalAck between outgoing packets and foreign acknowledgement of our packets, but that's a fairly natural combo imo